src/main/java/com/contrastsecurity/sdk/ContrastSDK.java

```diff
@@ -0,0 +1,327 @@
+
+            return IOUtils.toByteArray(is);
+        } finally {
+            IOUtils.closeQuietly(is);
+        }
+    }
+
+    /**
+     * Download a contrast.jar agent associated with this account. The user should save
+     * this byte array to a file named 'contrast.jar'. This signature takes a parameter
+     * which contains the name of the saved engine profile to download.
+     * <p>
+     * This uses 'default' as the profile name.
+     *
+     * @param type           the type of agent you want to download; Java, Java 1.5, .NET, or Node
+     * @param organizationId the ID of the organization,
+     * @return a byte[] array of the contrast.jar file contents, which the user should convert to a new File
+     * @throws IOException if there was a communication problem
+     */
+    public byte[] getAgent(AgentType type, String organizationId) throws IOException, UnauthorizedException {
+        return getAgent(type, organizationId, DEFAULT_AGENT_PROFILE);
+    }
+
+    public static void main(String[] args) throws UnauthorizedException, IOException, ResourceNotFoundException {
+        ContrastSDK conn = new ContrastSDK("username", "demo", "demo", LOCALHOST_API_URL);
+
+        String orgId = "add-your-org-id";
+        String appId = "add-your-app-id";
+
+        Gson gson = new Gson();
+
+        // System.out.println(gson.toJson(conn.getApplication(orgId, appId)));
+        // System.out.println(gson.toJson(conn.getApplications(orgId)));
+        // System.out.println(gson.toJson(conn.getCoverage(orgId, appId)));
+        // System.out.println(gson.toJson(conn.getTraces(orgId, appId)));
+        // System.out.println(gson.toJson(conn.getLibraries(orgId, appId)));
+        // FileUtils.writeByteArrayToFile(new File("contrast.jar"), conn.getAgent(AgentType.JAVA, orgId));
+    }
+
+    // ------------------------ Utilities -----------------------------------------------
+
+    public InputStream makeRequest(String method, String path) throws IOException, UnauthorizedException {
+        String url = restApiURL + path;
+        HttpURLConnection connection = makeConnection(url, method);
+        InputStream is = connection.getInputStream();
+        int rc = connection.getResponseCode();
+        if (rc >= BAD_REQUEST && rc < SERVER_ERROR) {
+            IOUtils.closeQuietly(is);
+            throw new UnauthorizedException(rc);
+        }
+        return is;
+    }
+
+    public HttpURLConnection makeConnection(String url, String method) throws IOException {
+        HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+        connection.setRequestMethod(method);
+        connection.setRequestProperty(RequestConstants.AUTHORIZATION, ContrastSDKUtils.makeAuthorizationToken(user, serviceKey));
+        connection.setRequestProperty(RequestConstants.API_KEY, apiKey);
+        connection.setUseCaches(false);
+        return connection;
+    }
+
+    private static final String GET_REQUEST = "GET";
```

Comment:
Use /contrast-sdk-java/src/main/java/com/contrastsecurity/http/HttpMethod.java instead of this harcoded strings